runner/src/simulate.rs

```diff
@@ -95,58 +118,391 @@ impl Scenario {
             | Self::CeramicWriteOnly
             | Self::CeramicNewStreams
             | Self::CeramicQuery
-            | Self::CeramicModelReuse => match peer {
-                Peer::Ceramic(peer) => Ok(peer.ceramic_addr.clone()),
-                Peer::Ipfs(_) => Err(anyhow!(
-                    "cannot use non ceramic peer as target for simulation {}",
-                    self.name(),
-                )),
-            },
+            | Self::CeramicModelReuse
+            | Self::EventIdSync => Ok(peer
+                .ceramic_addr()
+                .ok_or_else(|| {
+                    anyhow!(
+                        "cannot use non ceramic peer as target for simulation {}",
+                        self.name(),
+                    )
+                })?
+                .to_owned()),
         }
     }
 }
 
-#[tracing::instrument]
-pub async fn simulate(opts: Opts) -> Result<()> {
-    let mut metrics = Metrics::init(&opts)?;
+type MetricsCollector = Box<dyn MetricsCollection + Send + Sync>;
 
-    let peers: Vec<Peer> = parse_peers_info(opts.peers)
-        .await?
-        .into_iter()
-        .filter(|peer| matches!(peer, Peer::Ceramic(_)))
-        .collect();
+#[async_trait::async_trait]
+trait MetricsCollection: std::fmt::Debug {
```

Comment:
I probably should have attached this to the existing metrics struct.. would like to revisit to hopefully simplify/clean up the validation a bit, but might punt for now...